File: Leetcode/700-Search-In-Binary-Search-Tree/SearchInBinarySearchTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def searchBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        cur = root
        while cur:
            if cur.val == val:
                break
            if cur.val > val:
                cur = cur.left
                continue
            cur = cur.right
        return cur

        

File: Leetcode/700-Search-In-Binary-Search-Tree/SearchInBinarySearchTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode searchBST(TreeNode root, int val) {
        TreeNode cur = root;
        while (cur!=null) {
            if (cur.val == val) {
                break;
            }
            if (cur.val > val) {
                cur = cur.left;
                continue;
            }
            cur = cur.right;
        }
        return cur;
    }
}
